lib/Support/JitRunner.cpp

```diff
@@ -166,6 +171,35 @@ static LogicalResult convertAffineStandardToLLVMIR(ModuleOp module) {
   return manager.run(module);
 }
 
+static Error compileAndRunEntryPointFunction(
+    ModuleOp module, StringRef entryPoint,
+    std::function<llvm::Error(llvm::Module *)> transformer, void **data) {
+  // Create the JIT object cache that will be used to retrieve the JIT-compiled
+  // code and dump it to object file.
+  std::unique_ptr<MLIRObjectCache> objectCache;
+  if (dumpObjectFile)
+    objectCache = std::make_unique<MLIRObjectCache>();
+
+  SmallVector<StringRef, 4> libs(clSharedLibs.begin(), clSharedLibs.end());
+  auto expectedEngine = mlir::ExecutionEngine::create(module, transformer, libs,
+                                                      objectCache.get());
+  if (!expectedEngine)
+    return expectedEngine.takeError();
+
+  auto engine = std::move(*expectedEngine);
+  auto expectedFPtr = engine->lookup(entryPoint);
+  if (!expectedFPtr)
+    return expectedFPtr.takeError();
+
+  if (dumpObjectFile)
+    objectCache->dumpToObjectFile(inputFilename + ".o");
```

Comment:
It would be great to have the name configurable

Reply:
Added a flag for that